pkg/rpcservice/rpc_service.go

```diff
@@ -0,0 +1,111 @@
+package rpcservice
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/stripe/stripe-cli/rpc"
+)
+
+// Config provides the configuration for the RPC service.
+type Config struct {
+	// Port is the port number to listen to on localhost
+	Port int
+
+	// Info, error, etc. logger. Unrelated to API request logs.
+	Log *log.Logger
+}
+
+// RPCService implements the gRPC interface and starts the gRPC server.
+type RPCService struct {
+	cfg *Config
+
+	grpcServer *grpc.Server
+}
+
+// ConfigOutput is the config that clients will need to connect to the gRPC server. This is printed
+// out for clients to parse.
+type ConfigOutput struct {
+	// Address is the localhost address of the gRPC server
+	Address string `json:"address"`
+}
+
+// New creates a new RPC service
+func New(cfg *Config) *RPCService {
+	if cfg.Log == nil {
+		cfg.Log = &log.Logger{Out: ioutil.Discard}
+	}
+
+	grpcServer := grpc.NewServer(
+		grpc.UnaryInterceptor(serverUnaryInterceptor),
+		grpc.StreamInterceptor(serverStreamInterceptor),
+	)
+
+	return &RPCService{
+		cfg:        cfg,
+		grpcServer: grpcServer,
+	}
+}
+
+// Run starts a gRPC server on localhost
+func (srv *RPCService) Run(ctx context.Context) {
+	lis := srv.createListener()
+
+	srv.printConfig(ConfigOutput{
+		Address: lis.Addr().String(),
+	})
+
+	rpc.RegisterStripeCLIServer(srv.grpcServer, srv)
+
+	if err := srv.grpcServer.Serve(lis); err != nil {
+		srv.cfg.Log.Fatalf("Failed to serve gRPC server on %s: %v", lis.Addr().String(), err)
+	}
+}
+
+func (srv *RPCService) createListener() net.Listener {
+	address := "127.0.0.1:" // empty port => an available port is automatically chosen
```

Comment:
Since you use `net.Listen("tcp"` later, I think we should use `IPv6loopback` here instead of `127.0.0.1` (side point, it makes me sad that golang has a constant for IPv6 loopback but not IPv4)

Reply:
Updated with `IPv6loopback` and printing the host and port in separate fields:

![Screen Shot 2021-04-28 at 10 59 55 AM](https://user-images.githubusercontent.com/71457708/116453708-0c0e2d00-a814-11eb-892f-037891d094b5.png)